int32 publish_count
bool f_bump
bool b_bump
bool r_bump
bool l_bump

